### DISCUSSION TITLE
How to retrieve the size of the window (not the screen)?

### DISCUSSION THREAD

**Discussion by ghost** [MonoGame/MonoGame#8388] [General]
Hello,

I would like to retrive the new dimensions of the window each time my user resizes it. I have found a few interesting posts on the old forum, but every solution used the size of the screen, not the actual window. `GraphicsDeviceManager.PreferredBackBufferWidth/Height` only return the values I have set manually in the constructor, and do not change when the window is resized; and `GraphicsDevice.Adapter.CurrentDisplayMode` only returns the size of the screen.

I'm using the `Window.ClientSizeChanged` event to check for any changes, but unfortunately this event doesn't pass any useful info in its EventArgs.

**Comment by ghost** [comment]
Nevermind. I used the wrong keyword for my research. Using "viewport" and "resolution" gave me a much better result.


        
        private void OnClientSizeChangedCallback(object sender, EventArgs e)
        {
            Vector2 windowResolution = new(this.GraphicsDevice.Viewport.Width, this.GraphicsDevice.Viewport.Height);
            MouseInput mouseInput = InputManager.GetScheme<MouseInput>();
            mouseInput.WindowResolution = windowResolution;
        }

**Comment by Peewi** [comment]
Although it's not provided in the `GameWindow.ClientSizeChanged` event, you can get the window size from `GameWindow.ClientBounds`.

The sender of `GameWindow.ClientSizeChanged` appears to be the `GameWindow`.

**Reply by ThomasFOG** [reply]
Mind that ClientBounds might not exactly be what you think it is if the user has a high-DPI monitor with a scaling factor applied (and values will also differ depending on the system; e.g. both linux and macOS have completely different behavior than Windows in that regard).

**Reply by stromkos** [reply]
The removal of the taskbar / soft button space and client text zoom settings(yes DPI, bit different location), happens on some Linux desktops and configurations and not on others.

Bottom line is never consider most of the values returned as an accurate screen measurement, but an acceptable and visible start on all platforms.
